### PR TITLE
Add initial autonomous mode selection logic

### PR DESCRIPTION
**Pull request by aschokking**
@JohnGilb 
This depends on a SeriouslyCommonLib PR: https://github.com/Team488/SeriouslyCommonLib/pull/32

### CODE REVIEW

**Comment by aschokking** [comment]
This is mostly about how we build/select autonomous commands rather than what the actual commands will do.

File: CompetitionTemplate/src/competition/subsystems/autonomous/selection/BaseAutonomousModeSetCommand.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    
+    @Inject
+    public BaseAutonomousModeSetCommand(AutonomousModeSelector autonomousModeSelector) {
+        this.autonomousModeSelector = autonomousModeSelector;
+        this.requires(this.autonomousModeSelector);
+    }
+
+    @Override
+    public void execute() {
+        // no-op, all work done in initialize
```
Comment (JohnGilb) [inline CompetitionTemplate/src/competition/subsystems/autonomous/selection/BaseAutonomousModeSetCommand.java]: At first I thought that you had forgotten your init method here. =]

Modify the comment to something like: "...all work done in the initialize method of subclasses".

Comment (aschokking) [inline CompetitionTemplate/src/competition/subsystems/autonomous/selection/BaseAutonomousModeSetCommand.java]: addressed!

**Comment by JohnGilb** [comment]
lgtm

**Pull request merged**